### DISCUSSION TITLE
Transitioning to BTF Format for eBPF Maps in our examples

### DISCUSSION THREAD

**Discussion by kerwenwwer** [cilium/ebpf#1398] [Q&A]
Hi, when I reviewed the examples in this project, I noticed many of our examples utilize the older bpf_map_def method for declaring eBPF maps. However, since 2018, we have the new BTF format available for declaring eBPF maps, e.g.,
```C=
struct {
	__uint(type, BPF_MAP_TYPE_ARRAY);
	__type(key, __u32);
	__type(value, __u64);
	__uint(max_entries, 1);
} kprobe_map SEC(".maps");
```
At the same time, I've also discovered that even Linux 4.19 has reached its End of Life (EOL) as of January this year. I think, although forward compatibility is very important, for those who now wish to understand how to write and use eBPF from the examples, it might be more appropriate to default to examples based on the BPF Type Format. For users with older kernel versions, perhaps we could offer an additional guide to inform them on how to modify map declarations.

**Comment by lmb** [comment]
Makes sense. BTF style map definitions will work on old kernels as well, since it's all in user space. Feel free to submit a PR.
